django-stubs/core/mail/backends/base.pyi

```diff
@@ -13,4 +13,4 @@ class BaseEmailBackend:
     def __exit__(
         self, exc_type: Type[BaseException], exc_value: BaseException, traceback: types.TracebackType
     ) -> None: ...
-    def send_messages(self, email_messages: Iterable[EmailMessage]) -> int: ...
+    def send_messages(self, email_messages: List[EmailMessage]) -> int: ...
```

Comment:
Can you please explain this change? Why `Iterable` is not good enough?

Reply:
Moreover, `List` is invariant. It might cause problems.

Reply:
Given the way `send_messages` is invoked (https://github.com/django/django/blob/3.2.6/django/core/mail/__init__.py#L83-L87), it is more appropriate to type it as list-like object. This allows the implementation of subclasses to use methods like `__len__` and `__getitem__`, and overwritting the method with signature of `def send_messages(self, email_messages: Iterable[EmailMessage]) -> int` will still be acceptable.

Reply:
Since `List` is invariant, perhaps `Sequence` will be better here. The PR has been updated.